plugin/trino-mysql/src/test/java/io/trino/plugin/mysql/BaseMySqlConnectorTest.java

```diff
@@ -629,4 +635,43 @@ public void verifyMySqlJdbcDriverNegativeDateHandling()
             }
         }
     }
+
+    @Override
+    protected void createTableForWrites(String createTable, String tableName, Optional<String> primaryKey, OptionalInt updateCount)
+    {
+        super.createTableForWrites(createTable, tableName, primaryKey, updateCount);
+        primaryKey.ifPresent(key -> addPrimaryKey(createTable, tableName, key));
+    }
+
+    private void addPrimaryKey(String createTable, String tableName, String primaryKey)
+    {
+        Matcher matcher = Pattern.compile("CREATE TABLE .* \\(.*\\b" + primaryKey + "\\b\\s+([a-zA-Z0-9()]+).*", Pattern.CASE_INSENSITIVE).matcher(createTable);
+        if (matcher.matches()) {
+            String type = matcher.group(1).toLowerCase(Locale.ENGLISH);
+            if (type.contains("varchar") || type.contains("char")) {
+                // Mysql requires the primary keys must hava a fixed length, here use the 255 length that is just long enough for the test
```

Comment:
What happens if the length is insufficient?

Reply:
If there are exactly the same prefix(255) data and final in the case is try: 

- insert into target, then meet the `Duplicate entry xx for ...` error. 
- update/delete target, then test fails with wrong assert of update count and the expected data.